File: src/main/java/ninekothecat/catconomy/commands/balance/BalanceCommandExecutor.java
package ninekothecat.catconomy.commands.balance;

import ninekothecat.catconomy.Catconomy;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

import java.text.MessageFormat;

public class BalanceCommandExecutor implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, String[] args) {
        String shortPrefix = Catconomy.prefix.getShortPrefix();
        if (!sender.hasPermission("catconomy.balance")){
            sender.sendMessage(ChatColor.RED + "You are not authorised to do that");
            return false;
        }
        switch (args.length) {
            case 0:
                getSelfMoneyAmount(sender, shortPrefix);
                return true;
            case 1:
                getOtherPlayersMoneyAmount(sender, args, shortPrefix);
                return true;
            default:
                return false;
        }
    }

    private void getSelfMoneyAmount(CommandSender sender, String shortPrefix) {
        Player player = (Player) sender;
        final String MONEY_AMOUNT = Double.toString(Catconomy.getBalanceHandler().getBalance(player.getUniqueId()));
        sender.sendMessage(MessageFormat.format("{0}Your balance is {1} {2}",
                ChatColor.YELLOW,
                ChatColor.GOLD + MONEY_AMOUNT,
                ChatColor.AQUA + shortPrefix));
    }

    private void getOtherPlayersMoneyAmount(CommandSender sender, String[] args, String shortPrefix) {
        Player player = Catconomy.getPlayerFromName(args[0]);
        if (player != null && Catconomy.getBalanceHandler().userExists(player.getUniqueId())) {
            final String MONEY_AMOUNT = Double.toString(Catconomy.getBalanceHandler().getBalance(player.getUniqueId()));
            sender.sendMessage(MessageFormat.format("{0}{1}s balance is : {2}{3} {4}{5}",
                    ChatColor.YELLOW,
                    player.getDisplayName(),
                    ChatColor.GOLD,
                    MONEY_AMOUNT,
                    ChatColor.AQUA,
                    shortPrefix));
        } else {
            sender.sendMessage(MessageFormat.format("{0} Does not exist", ChatColor.GOLD + args[0]));
        }
    }
}

File: src/main/kotlin/ninekothecat/catconomy/commands/balance/BalanceCommandExecutor.kt
package ninekothecat.catconomy.commands.balance

import ninekothecat.catconomy.Catconomy
import ninekothecat.catplugincore.utils.player.getPlayerFromName
import ninekothecat.catplugincore.utils.sendError
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class BalanceCommandExecutor : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        val shortPrefix: String = Catconomy.prefix!!.shortPrefix
        if (!sender.hasPermission("catconomy.balance")) {
            sender.sendMessage("${ChatColor.RED}You are not authorised to do that")
            return false
        }
        return when (args.size) {
            0 -> if (sender is Player){
                    getSelfMoneyAmount(sender, shortPrefix)
                    true
                }else {
                    sendError("You can't check the balance of yourself because you aren't a player", sender)
                    false
                }
            1 -> {
                getOtherPlayersMoneyAmount(sender, args, shortPrefix)
                true
            }
            else -> false
        }
    }

    private fun getSelfMoneyAmount(sender: CommandSender, shortPrefix: String) {
        if (sender is Player) {
            val moneyAmount =
                Catconomy.getBalanceHandler()!!.getBalance(sender.uniqueId).toString()
            sender.sendMessage("${ChatColor.YELLOW}Your balance is ${ChatColor.GOLD}${moneyAmount} ${ChatColor.AQUA}${shortPrefix}")
        }else {
            sendError("You can't check your own balance, if you aren't a player",sender)
        }
    }

    private fun getOtherPlayersMoneyAmount(sender: CommandSender, args: Array<String>, shortPrefix: String) {
        val player = getPlayerFromName(args[0])
        if (player != null && Catconomy.getBalanceHandler()!!.userExists(player.uniqueId)) {
            val moneyAmount =
                Catconomy.getBalanceHandler()!!.getBalance(player.uniqueId).toString()
            sender.sendMessage("${ChatColor.YELLOW}${player.displayName}'s balance is : ${ChatColor.GOLD}${moneyAmount} ${ChatColor.AQUA}$shortPrefix")
        } else {
            sender.sendMessage("${ChatColor.GOLD}${args[0]} Does not exist")
        }
    }
}